src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java

```diff
@@ -65,22 +74,98 @@ public CCDCaseApi(
+            .map((caseMapper::from));
+    }
 
-        if (claims.size() > 1) {
+    public void linkDefendant(String externalId, String defendantId, String authorisation) {
+        User user = userService.authenticateAnonymousCaseWorker();
+        CaseDetails caseResult = searchByExternalId(authorisation, externalId)
+            .orElseThrow(() -> new CoreCaseDataStoreException("Claim not found by external ID"));
+
+        caseAccessApi.grantAccessToCase(user.getAuthorisation(),
+            authTokenGenerator.generate(),
+            user.getUserDetails().getId(),
+            JURISDICTION_ID,
+            CASE_TYPE_ID,
+            caseResult.getId() + "",
+            new UserId(defendantId)
+        );
+    }
+
+    public List<Claim> getByDefendantId(String id, String authorisation) {
+        User caseWorker = userService.authenticateAnonymousCaseWorker();
+        List<String> caseIdsGivenUserIdHasAccessTo = caseAccessApi.findCaseIdsGivenUserIdHasAccessTo(
+            caseWorker.getAuthorisation(),
+            authTokenGenerator.generate(),
+            caseWorker.getUserDetails().getId(),
+            JURISDICTION_ID,
+            CASE_TYPE_ID,
+            id
+        );
+
+        return caseIdsGivenUserIdHasAccessTo.stream()
+            .map((caseId) -> readCase(authorisation, caseId))
+            .filter((claim -> !claim.getSubmitterId().equals(id)))
+            .collect(Collectors.toList());
+    }
+
+    public Optional<Claim> getByLetterHolderId(String id, String authorisation) {
+        User user = userService.authenticateAnonymousCaseWorker();
+
+        List<String> letterHolderCases = caseAccessApi.findCaseIdsGivenUserIdHasAccessTo(
+            user.getAuthorisation(),
+            authTokenGenerator.generate(),
+            user.getUserDetails().getId(),
+            JURISDICTION_ID,
+            CASE_TYPE_ID,
+            id
+        );
+        if (letterHolderCases.size() > 1) {
+            throw new DefendantLinkingException("More than 1 case a letter holder ID has access to found");
+        }
+        if (letterHolderCases.size() == 0) {
+            throw new DefendantLinkingException("No cases found that the letter holder ID has access to");
+        }
+
+        return Optional.of(readCase(authorisation, letterHolderCases.get(0)));
+    }
+
+    private Claim readCase(String authorisation, String caseId) {
+        UserDetails userDetails = userService.getUserDetails(authorisation);
+        return caseMapper.from(
+            convertToCCDCase(
+                coreCaseDataApi.readForCitizen(
+                    authorisation,
+                    authTokenGenerator.generate(),
+                    userDetails.getId(),
+                    JURISDICTION_ID,
+                    CASE_TYPE_ID,
+                    caseId
+                ).getData()
+            )
+        );
+    }
+
+    private Optional<CaseDetails> searchByExternalId(String authorisation, String externalId) {
```

Comment:
We have two methods `searchByExternalId` and `getByExternalId` where the latter calls the former and additionally maps a CCD object to Claim Store domain object. Would you mind renaming the methods so that intents are expressed more clearly?